storage/shard.go

```diff
@@ -1535,6 +1541,10 @@ func (s *dbShard) FetchBlocksMetadataV2(
 					blockStart, err)
 			}
 
+			// Make sure ID and tags get cleaned up after read is done
+			ctx.RegisterFinalizer(id)
+			ctx.RegisterFinalizer(resource.FinalizerFn(tags.Close))
```

Comment:
should we just add `ctx.RegisterCloser(tags)` to avoid allocs like this?

Reply:
Sure thing.